_posts/2017-11-22-macros.md

```diff
@@ -0,0 +1,291 @@
+---
+category: blog
+by: Ólafur Páll Geirsson
+title: "What's happening with macros?"
+---
+
+In September, we posted a
+["Roadmap towards non-experimental macros"](http://www.scala-lang.org/blog/2017/10/09/scalamacros.html).
+It's been two months since then and I'd like to share an update on
+what's been happening.
+
+## A diverse ecosystem
+
+When starting this project, we wanted to first get an idea of how macros are
+used in the community.
+Not surprisingly, we discovered that macros are used in a variety of ways.
+Here are some popular use-cases we found
+
+- **code generation**: to avoid writing boilerplate by hand, which is
+  both error-prone and cumbersome. This is used by [Play JSON] and [Spark].
+- **embedded DSLs**: to attach new semantics to existing language syntax.
+  This includes a wide range of applications from database query
+  libraries like [Quill], to hardware construction frameworks like [Chisel] and
+  build tools like [sbt].
+- **optimizations**: to avoid expensive runtime allocations and method calls.
+  This includes libraries like [spire] and [log4s].
+- **developer ergonomics**: to surface source context information like variable
+  names, file names and line numbers. This includes libraries like 
+  [ScalaTest] and [sourcecode].
+- **improved type-safety**: to increase the static guarantees of your programs
+  beyond what the Scala language can provide. This includes libraries like [Refined].
+
+The current scala-reflect macro system has in fact done a phenomenal job
+at accommodating these diverse use-cases under a unified API.
+The Scala ecosystem rich as it is thanks to this wide range of applications
+that are enabled by scala-reflect macros.
+
+To get a better feel for how common each use-case is,
+we performed an analysis on a corpus of 20.906 source files containing ~3
```

Comment:
Should be: "20,906" not "20.906". 20.906 implies that you analyzed just over 20 files with a combined 3 million lines of code 😛